source/Registry.cpp

```diff
@@ -176,13 +200,25 @@ void Registry::join_with(const FieldModel& field_model) {
   }
 }
 
+void Registry::join_with(const LiteralModel& literal_model) {
+  auto iterator = literal_models_.find(literal_model.pattern());
```

Comment:
Maybe assert that the pattern is non-empty?

Reply:
Now using `mt_assert`:
```c++
const std::optional<std::string> pattern{literal_model.pattern()};
mt_assert(pattern);
```